apps/performance/default-onpush/src/app/directive/ngForEmpty.directive.ts

```diff
@@ -0,0 +1,28 @@
+import { NgFor } from '@angular/common';
+import { Directive, DoCheck, EmbeddedViewRef, Input, TemplateRef, ViewContainerRef, inject } from '@angular/core';
+
+@Directive({
+  selector: '[ngForEmpty]',
+  standalone: true,
+  hostDirectives: [
+    {
+      directive: NgFor,
+      inputs: ['ngForOf:ngForEmptyOf', 'ngForTrackBy:ngForEmptyTrackBy'],
+    }
+  ],
+})
+export class NgForEmptyDirective<T> implements DoCheck {
+  private vcr = inject(ViewContainerRef);
+  @Input() ngForEmptyOf: T[] | undefined;
+  @Input() ngForEmptyElse!: TemplateRef<any>;
+  private ref?: EmbeddedViewRef<unknown>;
+
+  ngDoCheck(): void {
+    this.ref?.destroy();
+    if (!this.ngForEmptyOf || this.ngForEmptyOf.length === 0) {
+      this.ref = this.vcr.createEmbeddedView(this.ngForEmptyElse);
+    }
+  }
+}
+
+export { NgForEmptyDirective as NgForEmpty };
```

Comment:
you can now use the new control flow to delete this custom directive 😜

Reply:
true, new control flow is better, on V17

Reply:
you can try it if you want, if you rebase your project and npm ci, you will be on angular v17